test/unit/CSV/CSVWriterTest.php

```diff
@@ -134,6 +136,55 @@ public function test_it_does_not_write_column_headers_before_subsequent_rows()
         fclose($file);
     }
 
+    public static function providerQuotedHeaders(): array
+    {
+        $row = ['our data' => 'is at times', 'really big' => 'often'];
+
+        return [
```

Comment:
Probably want a testcase with the default (e.g. `->open($file, [])`) as well as the explicit `['quote_headers' => true]` and `['quote_headers' => false]` cases?